src/components/headerSearch.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+
+import { useEffect, useRef, useState } from "react";
+import Image from "next/image";
+import { useRouter } from "next/navigation";
+import { Search } from "react-feather";
+
+import { getMoviesByTitle } from "@/services/movieService";
+
+type SearchResult = {
+  id: number;
+  title: string;
+  posterPath: string;
+  releaseDate: Date | null;
+};
+
+// PLACEHOLDER LOGIC: Currently returns movies that includes searched word in title
+// Searches at maximum every 1 seconds so search isn't spammed on every input change
+// "See all results" takes currently nowhere as there is no search page yet
+
+export const HeaderSearch = () => {
+  const [search, setSearch] = useState<string>("");
+  const [results, setResults] = useState<SearchResult[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const router = useRouter();
+  const searchBarRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const searchDelay = setTimeout(() => {
+      const getMovies = async () => {
+        if (search.trim() === "") {
+          setLoading(false);
+          return;
+        }
+        const movies = await getMoviesByTitle(search);
+        setResults(movies);
+        setLoading(false);
+      };
+
+      getMovies();
+    }, 1000);
+    return () => clearTimeout(searchDelay);
+  }, [search]);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (searchBarRef.current && !searchBarRef.current.contains(event.target as Node)) {
+        setSearch("");
+        setResults([]);
```

Comment:
Tarviiko näitä nollata?

Reply:
Resultseja ei tarvi, otan sen pois, setSearchin nollaus piilotaa sen dropdrownin.

Reply:
Eiku niin, nyt se input häviää kans kun se muu sulkeutuu. Pitää ehkä ihmetellä vielä vähän lisää ajatuksen kanssa 😄